library/std/src/sys/pal/unix/thread.rs

```diff
@@ -257,14 +257,31 @@ impl Thread {
         CString::new(name).ok()
     }
 
+    #[cfg(target_os = "haiku")]
+    pub fn get_name() -> Option<CString> {
```

Comment:
It took me quite a while to find the documentation for this. Could you add a link to https://www.haiku-os.org/legacy-docs/bebook/TheKernelKit_ThreadsAndTeams_Overview.html or some other reference?